core/dbt/parser/base.py

```diff
@@ -189,7 +189,7 @@ def _create_error_node(
         # message reasons
         return UnparsedNode(
             name=name,
-            resource_type=self.resource_type,
+            resource_type=self.resource_type,  # type: ignore
```

Comment:
It was complaining that the type here didn't match the type defined in UnparsedNode. I changed the type in UnparsedNode to NodeType, because changing the other direction didn't work because of abtractmethod signatures in the base parser class.